naive_ralloc/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+#![feature(allocator)]
+#![feature(linkage)]
+#![no_std]
+#![allocator]
+
+#[macro_use]
+extern crate sc;
+
+use core::ptr;
+
+// fs/mmap.c
+pub unsafe fn brk(brk: usize) -> usize {
+    syscall!(BRK, brk)
+}
+
+fn allocate(size: usize) -> *mut u8 {
+    let align = 16;
+    let mask = align - 1;
+    unsafe {
+        let cur = brk(0);
+        let aligned = (cur + mask) & !mask;
+        let new = aligned + size;
+        let actual = brk(new);
+        if actual < new {
+            // TODO: Should an allocator actually return 0?
+            return ptr::null_mut();
```

Comment:
I think it's UB to return NULL. I think it would be better just to abort on failures.